scripts/release-notes.py

```diff
@@ -269,40 +275,111 @@
 release_notes = {}
 missing_release_notes = []
 
+def collect_authors(commit):
+    authors = set()
+    author = author_aliases.get(commit.author.name, commit.author.name)
+    if author != 'GitHub':
+        authors.add(author)
+    author = author_aliases.get(commit.committer.name, commit.committer.name)
+    if author != 'GitHub':
+        authors.add(author)
+    for m in coauthor.finditer(commit.message):
+        aname = m.group('name').strip()
+        author = author_aliases.get(aname, aname)
+        authors.add(author)
+    return authors
+
+
 def extract_release_notes(pr, title, commit):
+    authors = collect_authors(commit)
     if norelnote.search(commit.message) is not None:
         # Explicitly no release note. Nothing to do.
-        return
-
-    item = {'author': (commit.author.name, commit.author.email),
+        # Just report the author(s).
+        return None, authors
+
+    msglines = commit.message.split('\n')
+    curnote = []
+    innote = False
+    foundnote = False
+    cat = None
+    notes = []
+    for line in msglines:
+        m = coauthor.search(line)
+        if m is not None:
+            # We ignore co-authored-by lines.
+            continue
+        m = fixannot.search(line)
+        if m is not None:
+            # Fix/Close etc. Ignore.
+            continue
+
+        m = relnote.search(line)
+        if m is None:
+            # Current line does not contain a release note separator.
+            # If we were already collecting a note, continue collecting it.
+            if innote:
+                curnote.append(line)
+            continue
+
+        # We have a release note boundary. If we were collecting a
+        # note already, complete it.
+        if innote:
+            notes.append((cat, curnote))
+            curnote = []
+            innote = False
+
+        # Start a new release note.
+
+        firstline = m.group('note').strip()
+        if firstline.lower() == 'none':
+            # Release note: none - there's no note yet.
+            continue
+        foundnote = True
```

Comment:
should this line be above the previous block?  `Release note: None` is still a note, no?